pipelines/utilities/junit_to_qase.py

```diff
@@ -0,0 +1,274 @@
+
+        test_suite = test_case.get("classname")
+        test_result['suite'] = test_suite.split(".")[-1]
+
+        test_result['case'] = test_case.get("name")
+
+        skipped = test_case.findChildren("skipped", recursive=False)
+        if len(skipped) > 0:
+            test_result['skipped'] = skipped[0].get("message")
+
+        failure = test_case.findChildren("failure", recursive=False)
+        if len(failure) > 0:
+            test_result['failure'] = failure[0].get("message")
+            test_result['failure'] += '\n\n' + failure[0].get_text()
+
+        test_results.append(test_result)
+
+    return test_results
+
+
+def get_suite_id(suite_name):
+
+    print(f"getting suite {suite_name}")
+
+    query_string = urlencode({ "search": suite_name })
+    headers = {
+        "accept": "application/json",
+        "Token": qase_token
+    }
+    url = f"https://api.qase.io/v1/suite/{qase_project}?{query_string}"
+
+    resp = requests.get(url, headers=headers)
+    res = resp.json()
+    if res["result"]["count"] > 0:
+        id = res["result"]["entities"][0]["id"]
+        print(f"got suite {suite_name} with id {id}")
+        return id
+    else:
+        print(f"missed suite {suite_name}")
+        return None
+
+
+def add_missing_suite(suite_name):
+
+    print(f"adding missing suite {suite_name}")
+
+    payload = {
+        "title": suite_name,
+        "parent_id": parent_suite_id
+    }
+    headers = {
+        "accept": "application/json",
+        "content-type": "application/json",
+        "Token": qase_token
+    }
+    url = f"https://api.qase.io/v1/suite/{qase_project}"
+
+    resp = requests.post(url, json=payload, headers=headers)
+    res = resp.json()
+    print(f"added suite {suite_name} with id {res['result']['id']}")
+    return res["result"]["id"]
+
+
+def get_suites_id_and_add_missing_suites(test_results):
+    for result in test_results:
+        result["suite_id"] = get_suite_id(result["suite"])
+        if not result["suite_id"]:
+            result["suite_id"] = add_missing_suite(result["suite"])
+
+
+def get_case_id(case_name):
+
+    print(f"getting case {case_name}")
+
+    query_string = urlencode({ "search": case_name })
+    headers = {
+        "accept": "application/json",
+        "Token": qase_token
+    }
+    url = f"https://api.qase.io/v1/case/{qase_project}?{query_string}"
+
+    resp = requests.get(url, headers=headers)
+    res = resp.json()
+    if res["result"]["count"] > 0:
+        id = res["result"]["entities"][0]["id"]
+        print(f"got case {case_name} with id {id}")
+        return id
+    else:
+        print(f"missed case {case_name}")
+        return None
+
+
+def add_missing_case(case_name, parent_suite_id):
+
+    print(f"adding missing case {case_name} under parent suite {parent_suite_id}")
+
+    if len(case_name) > 128:
+        case_name = case_name[:127]
+
+    payload = {
+        "title": case_name,
+        "suite_id": parent_suite_id,
+        "automation": 2
```

Comment:
There won't be any description of the test case which is getting added here?

Reply:
No. There is no description section for a test case in a junit report.

Reply:
Ok, in that case, should we rely on syncing the test case from github to Qase like manual test case syncing?

Reply:
We don't have a better approach for this right now, but we can figure it out in the future.